--- sane-backends-1.0.22/backend/canon_dr.c	2011-01-30 10:49:46.000000000 -0800
+++ /var/tmp/sane-backends-1.0.22/backend/canon_dr.c	2011-08-15 14:20:29.000000000 -0700
@@ -3259,16 +3259,16 @@
     }
 
     /* eject paper leftover*/
-    if(object_position (s, SANE_FALSE)){
-      DBG (5, "sane_start: ERROR: cannot eject page\n");
-    }
+    //    if(object_position (s, SANE_FALSE)){
+    //      DBG (5, "sane_start: ERROR: cannot eject page\n");
+    //    }
 
     /* wait for scanner to finish eject */
-    ret = wait_scanner (s);
-    if (ret != SANE_STATUS_GOOD) {
-      DBG (5, "sane_start: ERROR: cannot wait scanner\n");
-      goto errors;
-    }
+    //    ret = wait_scanner (s);
+    //    if (ret != SANE_STATUS_GOOD) {
+    //      DBG (5, "sane_start: ERROR: cannot wait scanner\n");
+    //      goto errors;
+    //    }
 
     /* load the brightness/contrast lut with linear slope for calibration */
     ret = load_lut (s->lut, 8, 8, 0, 255, 0, 0);
@@ -3358,21 +3358,21 @@
     }
 
     /* grab next page */
-    ret = object_position (s, SANE_TRUE);
-    if (ret != SANE_STATUS_GOOD) {
-      DBG (5, "sane_start: ERROR: cannot load page\n");
-      goto errors;
-    }
+    //    ret = object_position (s, SANE_TRUE);
+    //    if (ret != SANE_STATUS_GOOD) {
+    //      DBG (5, "sane_start: ERROR: cannot load page\n");
+    //      goto errors;
+    //    }
 
     /* wait for scanner to finish load */
-    ret = wait_scanner (s);
-    if (ret != SANE_STATUS_GOOD) {
-      DBG (5, "sane_start: ERROR: cannot wait scanner\n");
-      goto errors;
-    }
+    //    ret = wait_scanner (s);
+    //    if (ret != SANE_STATUS_GOOD) {
+    //      DBG (5, "sane_start: ERROR: cannot wait scanner\n");
+    //      goto errors;
+    //    }
 
     /* start scanning */
-    ret = start_scan (s,0);
+    ret = start_scan (s,0xfe);
     if (ret != SANE_STATUS_GOOD) {
       DBG (5, "sane_start: ERROR: cannot start_scan\n");
       goto errors;
@@ -3412,16 +3412,16 @@
 
       /* big scanners and small ones in non-buff mode: OP to detect paper */
       if(s->always_op || !s->buffermode){
-        ret = object_position (s, SANE_TRUE);
-        if (ret != SANE_STATUS_GOOD) {
-          DBG (5, "sane_start: ERROR: cannot load page\n");
-          goto errors;
-        }
+	//        ret = object_position (s, SANE_TRUE);
+	//        if (ret != SANE_STATUS_GOOD) {
+	//          DBG (5, "sane_start: ERROR: cannot load page\n");
+	//          goto errors;
+	//        }
 
         /* user wants unbuffered scans */
         /* send scan command */
         if(!s->buffermode){
-          ret = start_scan (s,0);
+          ret = start_scan (s,0xfe);
           if (ret != SANE_STATUS_GOOD) {
             DBG (5, "sane_start: ERROR: cannot start_scan\n");
             goto errors;
