--- orig/sane-backends-1.0.22/backend/genesys.c	2011-02-13 15:10:18.000000000 -0800
+++ sane-backends-1.0.22/backend/genesys.c	2011-07-25 17:16:38.000000000 -0700
@@ -5746,6 +5746,15 @@
     }
 
   s->params.lines = ((br_y - tl_y) * s->dev->settings.yres) / MM_PER_INCH;
+
+  /* HACK. Max out 20 bit hardware register. */
+  if (br_y == 111) {
+    if (s->params.format == SANE_FRAME_RGB)
+      s->params.lines = ((1 << 20) - 1) / 3;
+    else
+      s->params.lines = (1 << 20) - 1;
+  }
+
   s->params.pixels_per_line =
     ((br_x - tl_x) * resolution) / MM_PER_INCH;

--- orig/sane-backends-1.0.22/backend/genesys_gl124.c	2011-01-15 17:01:28.000000000 -0800
+++ sane-backends-1.0.22/backend/genesys_gl124.c	2011-07-25 17:16:38.000000000 -0700
@@ -1727,7 +1727,7 @@
 				    depth) / 8));
 
 
-  dev->read_bytes_left = bytes_per_line * lincnt;
+  dev->read_bytes_left = (size_t) bytes_per_line * lincnt;
 
   DBG (DBG_info,
        "gl124_init_scan_regs: physical bytes to read = %lu\n",
@@ -1755,7 +1755,7 @@
       channels;
   else
     dev->total_bytes_to_read =
-      dev->settings.pixels * dev->settings.lines * channels * (depth / 8);
+      (size_t) dev->settings.pixels * dev->settings.lines * channels * (depth / 8);
 
   DBG (DBG_info, "gl124_init_scan_regs: total bytes to send = %lu\n",
        (u_long) dev->total_bytes_to_read);
